compiler/src/dotty/tools/dotc/CompilationUnit.scala

```diff
@@ -39,13 +40,21 @@ object CompilationUnit {
     assert(!unpickled.isEmpty, unpickled)
     val unit1 = new CompilationUnit(source)
     unit1.tpdTree = unpickled
-    if (forceTrees)
+    if (forceTrees) {
+      val force = new Force
       force.traverse(unit1.tpdTree)
+      unit1.containsQuotesOrSplices = force.containsQuotes
+    }
     unit1
   }
 
   /** Force the tree to be loaded */
-  private object force extends TreeTraverser {
-    def traverse(tree: Tree)(implicit ctx: Context): Unit = traverseChildren(tree)
```

Comment:
I don't understand why the change here is necessary. `forceTrees` is true only in the IDE,, right? But in the IDE we never get to ReifyQuotes.

Reply:
No, force tree is also used when compiling from TASTY. 
This covers the case when we recompile from TASTY a file with a quote and no splices.